pkg/kubelet/kubelet_pods.go

```diff
@@ -1359,6 +1359,15 @@ func (kl *Kubelet) generateAPIPodStatus(pod *v1.Pod, podStatus *kubecontainer.Po
 	spec := &pod.Spec
 	allStatus := append(append([]v1.ContainerStatus{}, s.ContainerStatuses...), s.InitContainerStatuses...)
 	s.Phase = getPhase(spec, allStatus)
+	// Check for illegal phase transition
```

Comment:
should we put this block before the getPhase call, and only call getPhase if this does not get executed?

i feel like the core bug is in getPhase with some garbage collection scenario.

Reply:
I would like to call `getPhase()` regardless so that we can catch it in the act and hopefully determine the root cause and possibly remove this check in the future.

Reply:
so i am skeptical that the logging will help identify the core problem any better.

i think in a future pr, we should look to augment getPhase to pass in what we think the computed phase should have been in the case of a terminal state to help understand or better isolate why its coming up with something different, but for now, this is fine and helps alleviate a major issue we are seeing.